components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/RESTAPIAdminClient.java

```diff
@@ -102,21 +108,112 @@ public void updateApi(APITemplateBuilder builder, String tenantDomain) throws Ax
 	}
 
 	/**
-	 * Delete the WebApp from Gateway
+	 * Deletes versioned web app configuration from the gateway
+	 *
 	 * @param tenantDomain
 	 * @throws AxisFault
 	 */
-	public void deleteApi(String tenantDomain) throws AxisFault {
+	public void deleteVersionedWebApp(String tenantDomain) throws AxisFault {
 		try {
 			if (tenantDomain != null && !("").equals(tenantDomain) &&
-			    !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+					!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
 				restApiAdminStub.deleteApiForTenant(qualifiedName, tenantDomain);
 			} else {
 				restApiAdminStub.deleteApi(qualifiedName);
 			}
-			
+
 		} catch (Exception e) {
 			throw new AxisFault("Error while deleting WebApp", e);
 		}
 	}
+
+	/**
+	 * Adds non-versioned web app configuration to the gateway
```

Comment:
fixed in commit 818ca74c18433a3a2ffa7a6f80e8ef2b09fffa70 of PR #153 for all methods in this class